executor/logging.go

```diff
@@ -7,15 +7,24 @@ import (
 )
 
 // bindLoggingPipe spawns a goroutine for passing through logging of the given output pipe.
-func bindLoggingPipe(name string, pipe io.Reader, output io.Writer) {
+func bindLoggingPipe(name string, pipe io.Reader, output io.Writer, logPrefix bool) {
 	log.Printf("Started logging %s from function.", name)
 
 	scanner := bufio.NewScanner(pipe)
-	logger := log.New(output, log.Prefix(), log.Flags())
+	logFlags := log.Flags()
+	if !logPrefix {
+		logFlags = 0
+	}
+
+	logger := log.New(output, log.Prefix(), logFlags)
```

Comment:
this feels a little odd, reusing the global logger `Prefix` here _even_ if `logPrefix` is false? It seems like we have a pretty clear indicator to set this to the empty string here instead of inheriting from the global logger `""`

Why not this, create the default logger and then use the setters if needed
```suggestion
	logger := log.New(output, log.Prefix(), log.Flags())
	if !logPrefix {
		logger.SetPrefix("")
		logger.SetFlags(0)
	}
```

Reply:
I am not sure that I follow the comment Lucas.

How is this affecting / mutating the global logger? From my perspective (it sounds like I might be wrong here?) I am only altering the new logger being created.

Reply:
I am not clear how the suggestion is better - one creates the "correct" logger fit for purpose. The other creates the logger, then mutates it.

Reply:
The prefix itself doesn't output any text when the flags are set to "0"

Reply:
Hopefully this explains it the approach?

If there's a functional reason why this is wrong, let me know and I'll revert it.

Reply:
@alexellis my point is that you are referencing something `log.Prefix()` which could for some reason change.  Additionally, you use it to set the prefix value _before_ inspecting the `logPrefix` value. In theory, the `log.SetPrefix` could be used somewhere else and this part of the code would then reflect that change, regardless of the value of `logPrefix`. The code simply doesn't read correctly for the intention of the function signature. We know _exectly_ what the log prefix value should be when `!logPrefix`. my code suggestion does exactly that, when `!logPredfix`, then it expeclitly sets the prefix to empty and flags to 0 

If you look at the log package source, the log line formatting happens here http://golang.org/src/log/log.go?s=9636:9664#L101, importantly when the flag is zero, the prefix value is simply the first thing put into the buffer. Setting the flags to zero wont actually stop the prefix value from being prefixed, the flags control the format of the  timestamp and where the `log.Prefix()` value is printed, if it is non-empty.

Reply:
It seems like a moot point because there is no other code in the watchdog which will change the log prefix. This was manually verified, however if you are particularly concerned then I'll explicitly set it to an empty string, which is what I did originally before realising it was unnecessary complication.

Reply:
Hopefully this satisfies the concern you raised. https://github.com/openfaas/of-watchdog/commit/c72f1f9eb0cacc69c20f9ec423ceeb069584fc04